### PR TITLE
Update modules to support latest versions of Coq and MathComp

### PR DESCRIPTION
**Pull request by yasunariw**
This PR imports some changes from [`pnp`](https://github.com/ilyasergey/pnp).
- Add a new external dependency ([`fcsl-pcm`](https://github.com/imdea-software/fcsl-pcm))
- Update the versions of existing dependencies
- Support installation by [opam](https://opam.ocaml.org/)

The dependency changes make the following modules redundant, so they will be removed.
- `Core/finmap.v`
- `Core/heap.v`
- `Core/ordtype.v`
- `Core/pcm.v`
- `Core/unionmap.v`

### CODE REVIEW

File: coq-htt.opam [hunk 1]
```diff
@@ -0,0 +1,35 @@
+opam-version: "2.0"
+version: "dev"
+
+maintainer: [
+  "Ilya Sergey <ilya.sergey@yale-nus.edu.sg>"
+  "Yasunari Watanabe <yasunari@comp.nus.edu.sg>"
```
[the remark this answers is not in the document]
Comment (clayrat) [reply coq-htt.opam:6]: Hmm I guess I should at least read this repo :)

File: coq-htt.opam [hunk 2]
```diff
@@ -2,17 +2,18 @@ opam-version: "2.0"
 version: "dev"
 
 maintainer: [
-  "Ilya Sergey <ilya.sergey@yale-nus.edu.sg>"
-  "Yasunari Watanabe <yasunari@comp.nus.edu.sg>"
+  "Aleks Nanevski"
+  "Alex Gryzlov"
+  "Germán Andrés Delbianco"
```
Comment (germanD) [inline coq-htt.opam:7]:
```suggestion
  "Aleks Nanevski <aleks.nanevski@imdea.org>"
  "Alex Gryzlov <alex.gryzlov@imdea.org>"
  "Germán Andrés Delbianco <german@nomadic-labs.com>"
```

File: coq-htt.opam [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+]
+authors: [
+  "Aleks Nanevski"
+  "Ilya Sergey"
+  "Germán Andrés Delbianco"
+]
+homepage: ["https://github.com/imdea-software/htt"]
```
Comment (nzyuzin) [inline coq-htt.opam:14]: There's the project page http://software.imdea.org/~aleks/htt/, which could be specified as a home page.

File: coq-htt.opam (hunk 2 above)
Comment (nzyuzin) [reply coq-htt.opam:7]: > "Alex Gryzlov <alex.gryzlov@imdea.org>"
The right email is: aliaksandr.hryzlou@imdea.org.

**Comment by nzyuzin** [comment]
Thanks for taking care to update the dependencies and connecting to opam!

File: coq-htt.opam (hunk 3 above)
Comment (ilyasergey) [reply coq-htt.opam:14]: It doesn't seems to be updated frequently.

Comment (nzyuzin) [reply coq-htt.opam:14]: Indeed it doesn't. It does provide some description of the system and the list of relevant publications though, which might give some idea of the system compared to the empty description in this repo.

Maybe that information can be referenced/moved into README.md (not necessarily in this PR of course).

Comment (germanD) [reply coq-htt.opam:14]: Right. We should better leave the pkg as it is, and then update the README (and replace @aleksnanevski  website with a project github' page). I will create an issue later and assign it to @clayrat

File: coq-htt.opam (hunk 2 above)
Comment (germanD) [reply coq-htt.opam:7]: My bad

**Comment by germanD** [comment]
LGTM!

File: coq-htt.opam (hunk 2 above)
Comment (germanD) [reply coq-htt.opam:7]: @ilyasergey can you apply the patch?

**Pull request merged**